File: src/app/brain/8227bd2f-d151-47ef-b78a-efd93f851d25/scratch/extract_layers.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('d:/web/golden/public/relax.json', 'utf8'));

function findLayers(obj) {
    let layers = [];
    if (typeof obj === 'object' && obj !== null) {
        if (Array.isArray(obj)) {
            obj.forEach(item => {
                layers = layers.concat(findLayers(item));
            });
        } else {
            if (obj.layers) {
                obj.layers.forEach(l => {
                    layers.push(l.nm || 'unnamed');
                });
            }
            Object.values(obj).forEach(v => {
                layers = layers.concat(findLayers(v));
            });
        }
    }
    return layers;
}

const allNames = Array.from(new Set(findLayers(data)));
console.log(allNames.join(', '));

File: src/app/brain/8227bd2f-d151-47ef-b78a-efd93f851d25/scratch/extract_layers.py
import json
with open('d:/web/golden/public/relax.json', 'r') as f:
    data = json.load(f)

def find_layers(obj):
    layers = []
    if isinstance(obj, dict):
        if 'layers' in obj:
            for l in obj['layers']:
                layers.append(l.get('nm', 'unnamed'))
        for v in obj.values():
            layers.extend(find_layers(v))
    elif isinstance(obj, list):
        for item in obj:
            layers.extend(find_layers(item))
    return layers

print(list(set(find_layers(data))))
